static/src/javascripts/projects/common/modules/ui/objectfit-polyfill.js

```diff
@@ -0,0 +1,39 @@
+define([
+    'common/utils/fastdom-promise',
+    'common/utils/$'
+], function (
+    fastdomPromise,
+    $
+) {
+    // Polyfill for Object-fit: https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit
+
+    function setBackgroundImage(src) {
+        return fastdomPromise.write(function () {
+            $('.js-object-fit-polyfill').each(function () {
+                var $container = $(this);
+
+                if (src) {
+                    $container
+                        .css('backgroundImage', 'url(' + src + ')')
+                        .addClass('immersive-main-media--fallback');
+                }
+            });
+        });
+    }
+
+    function init() {
+        return fastdomPromise.read(function () {
```

Comment:
I don't think you need to use fastdom here because  you're only getting the element, not querying anything that would force layout. You're also tying your polyfill directly to the immersive functionality here, maybe it would be better to pass a class name to the init and loop over the elements with that class name, inserting a wrapping element directly (rather than applying a class to something else)?